rest/src/main/kotlin/service/RestService.kt

```diff
@@ -10,18 +10,17 @@ import kotlin.contracts.contract
 public abstract class RestService(@PublishedApi internal val requestHandler: RequestHandler) {
 
     @PublishedApi
-    internal suspend inline fun <T> call(route: Route<T>, builder: RequestBuilder<T>.() -> Unit = {}): T {
+    internal suspend inline fun <T> call(
+        route: Route<T>,
+        baseUrl: String = Route.baseUrl,
+        builder: RequestBuilder<T>.() -> Unit = {},
+    ): T {
         contract {
             callsInPlace(builder, InvocationKind.EXACTLY_ONCE)
         }
-        val request = RequestBuilder(route)
+        val requestBuilder = RequestBuilder(route)
             .apply(builder)
-            .apply {
-                if (route.requiresAuthorizationHeader) {
-                    unencodedHeader(Authorization, "Bot ${requestHandler.token}")
-                }
-            }
-            .build()
-        return requestHandler.handle(request)
+        val interceptedBuilder = requestHandler.intercept(requestBuilder)
+        return requestHandler.handle(interceptedBuilder.build())
```

Comment:
```suggestion
        val interceptedBuilder = with(requestHandler) { requestBuilder.intercept() }
        return requestHandler.handle(interceptedBuilder.build())
```
also change this if `intercept` would get a member extension